internal/csi-addons/rbd/volumegroup.go

```diff
@@ -76,22 +76,32 @@ func (vs *VolumeGroupServer) RegisterService(server grpc.ServiceRegistrar) {
 //
 // Implementation steps:
 // 1. resolve all volumes given in the volume_ids list (can be empty)
-// 2. create the Volume Group
-// 3. add all volumes to the Volume Group
+// 2. check if the volumes belong to a (and all the same) group
+// 3. create the Volume Group
+// 4. verify that the Volume Group contains all the images (if it pre-exists)
+// 5. add all volumes to the Volume Group
 //
 // Idempotency should be handled by the rbd.Manager, keeping this function and
 // the potential error handling as simple as possible.
+//
+//nolint:gocyclo,cyclop // FIXME: make this function simpler
 func (vs *VolumeGroupServer) CreateVolumeGroup(
 	ctx context.Context,
 	req *volumegroup.CreateVolumeGroupRequest,
 ) (*volumegroup.CreateVolumeGroupResponse, error) {
+	groupName := req.GetName()
+
 	mgr := rbd.NewManager(vs.driverInstance, req.GetParameters(), req.GetSecrets())
 	defer mgr.Destroy(ctx)
 
 	// resolve all volumes
 	volumes := make([]types.Volume, len(req.GetVolumeIds()))
 	defer func() {
 		for _, vol := range volumes {
+			if vol == nil {
+				continue
+			}
```

Comment:
@nixpanic we dont get to this state isnt it?

Reply:
Not normally, but we can. The commit that adds this part has the following commit message:

> When an incorrect volumeID is passed while creating a VolumeGroup, the
> `.Destroy()` function causes a panic. With this extra check to skip
> `nil` volumes, the panic does not happen anymore.

Reply:
This can probably also be solved by creating an empty `volumes` slice, and appending each volume after it has been resolved. If that is something you prefer, I can take that approach.

Reply:
that makes sense, we could avoid this check

Reply:
Updated, please check again.